pom.xml

```diff
@@ -97,7 +97,9 @@
     <fasterxml.spark3.version>2.10.0</fasterxml.spark3.version>
     <kafka.version>2.0.0</kafka.version>
     <kafka.spark3.version>2.4.1</kafka.spark3.version>
-    <pulsar.version>2.8.1</pulsar.version>
+    <pulsar.version>2.10.1</pulsar.version>
```

Comment:
do the pulsar callbacks or existing functionality around pulsar still work?

Reply:
Will be reverting this back, since client-dep turned out not to be required for connector to work (it actually breaks it if included)